src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -31,44 +34,66 @@ import { ConfigService } from "../../config/config.service";
   templateUrl: "./navigation.component.html",
   styleUrls: ["./navigation.component.scss"],
 })
+@UntilDestroy()
 export class NavigationComponent {
+  activeElement: string;
   /** name of config array in the config json file */
   private readonly CONFIG_ID = "navigationMenu";
   /** all menu items to be displayed */
   public menuItems: MenuItem[] = [];
 
   constructor(
     private adminGuard: AdminGuard,
-    private configService: ConfigService
+    private configService: ConfigService,
+    private router: Router
   ) {
     this.configService.configUpdated.subscribe(() =>
       this.initMenuItemsFromConfig()
     );
+    this.router.events
+      .pipe(
+        startWith(new NavigationEnd(0, this.router.url, "")),
+        filter((event) => event instanceof NavigationEnd),
+        map((event: NavigationEnd) =>
+          // conservative filter matching all items that could fit to the given url
+          this.menuItems.filter((item) => event.url.startsWith(item.link))
+        )
+      )
+      .subscribe((items) => {
+        if (items.length === 0) {
+          this.activeElement = "";
+        } else if (items.length === 1) {
+          this.activeElement = items[0].name;
+        } else {
+          // If there are multiple matches (multiple elements start with the give link),
+          // use the element where the length is bigger. For example:
+          // '/attendance' matches both manage attendance ('/attendance') as well as
+          // '/attendance/add/day' (add day attendance).
+          // If 'attendance/add/day' was the active link, then only this link
+          // would have been filtered
+          this.activeElement = items.reduce((i1, i2) =>
+            i1.link.length > i2.link.length ? i1 : i2
```

Comment:
I think the _shorter_ element should be used? In the example with `/attendance`, the `/attendance` menu item should be highlighted and not the `/attendance/add/day`.

Reply:
This is counterintuitive but correct. The function matches the `url` with the `link`. So suppose the `url` is `/attendance`, `/attendance/add/day` wouldn't be matched (since `attendance` doesn't start with `attendance/add/day`).
The reason to choose this approach (as compared to the approach where the `link` would be matched against the `url`) is that in the latter approach, the url `/child/7` (child -> details: childId = 7) would not match against the link `/child`, since `/child/7` is not a prefix of `/child`. You can have a look at the test cases to verify that this is correct.

Reply:
Ok I understand it now, this makes sense. Thanks for the explanation.